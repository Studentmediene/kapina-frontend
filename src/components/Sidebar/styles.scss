@import 'variables.scss';

$social-media-button-sidebar: 2 * (2 * $social-media-button-width + 6 * $social-media-button-margin);

.sidebar {
  position: fixed;
  left: calc(50% + #{$breakpoint-large} / 2);
  bottom: $player-height;
  margin-bottom: $social-media-button-margin;
  margin-left: 5px;
  display: none;
  flex-direction: row;
  justify-content: flex-start;
}

@media screen and (min-width: $breakpoint-large + $social-media-button-sidebar) {
  .sidebar {
    display: flex;
  }
}
