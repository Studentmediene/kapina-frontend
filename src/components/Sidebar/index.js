import React from 'react';

import styles from './styles.scss';
import facebookLogo from 'components/common/assets/facebook.svg';
import instagramLogo from 'components/common/assets/instagram.svg';
import SocialMediaButton from 'components/common/button/SocialMediaButton';

const Sidebar = () => (
  <div className={styles.sidebar}>
    <SocialMediaButton
      link="https://www.facebook.com/radiorevolt.no/"
      image={facebookLogo}
      text="Facebook"
    />
    <SocialMediaButton
      link="https://www.instagram.com/radiorevolt/"
      image={instagramLogo}
      text="Instagram"
    />
  </div>
);

export default Sidebar;
