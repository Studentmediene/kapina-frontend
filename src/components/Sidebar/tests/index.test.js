import React from 'react';
import { shallow } from 'enzyme';

import Sidebar from '../';

describe('<Sidebar />', () => {
  it('renders correctly', () => {
    const tree = shallow(<Sidebar />);
    expect(tree).toMatchSnapshot();
  });
});
