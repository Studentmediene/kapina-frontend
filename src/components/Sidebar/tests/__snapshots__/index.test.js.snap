// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> renders correctly 1`] = `
<div
  className="sidebar"
>
  <SocialMediaButton
    image="facebook.svg"
    link="https://www.facebook.com/radiorevolt.no/"
    text="Facebook"
  />
  <SocialMediaButton
    image="instagram.svg"
    link="https://www.instagram.com/radiorevolt/"
    text="Instagram"
  />
</div>
`;
